storage/rocksdb/ha_rocksdb.cc

```diff
@@ -15076,6 +15118,373 @@ void rdb_tx_multi_get(Rdb_transaction *tx,
   tx->multi_get(column_family, num_keys, keys, values, statuses, sorted_input);
 }
 
+////
+/****************************************************************************
+ * DS-MRR implementation
+ ***************************************************************************/
+
+/*
+  A MultiGet-MRR is applicable if
+  - using the clustered PK
+  - all ranges are single-point lookups using full PK.
+*/
+ha_rows ha_rocksdb::multi_range_read_info_const(uint keyno, RANGE_SEQ_IF *seq,
+                                                void *seq_init_param,
+                                                uint n_ranges, uint *bufsz,
+                                                uint *flags,
+                                                Cost_estimate *cost) {
+  ha_rows res;
+  THD *thd = table->in_use;
+
+  // MultiGet-MRR is allowed only with these settings:
+  //  optimizer_switch='mrr=on,mrr_cost_based=off'
+
+  bool mrr_enabled =
+      thd->optimizer_switch_flag(OPTIMIZER_SWITCH_MRR) &&
+      !thd->optimizer_switch_flag(OPTIMIZER_SWITCH_MRR_COST_BASED);
+
+  res = handler::multi_range_read_info_const(keyno, seq, seq_init_param,
+                                             n_ranges, bufsz, flags, cost);
+
+  if (res == HA_POS_ERROR || m_lock_rows != RDB_LOCK_NONE || !mrr_enabled)
+    return res;
+
+  if (keyno == table->s->primary_key) {
+    bool all_eq_ranges = true;
+    KEY_MULTI_RANGE range;
+    range_seq_t seq_it;
+    seq_it = seq->init(seq_init_param, n_ranges, *flags);
+    while (!seq->next(seq_it, &range)) {
+      if (!(range.range_flag & UNIQUE_RANGE)) {
```

Comment:
Could you check if thd is marked as killed or not in the loop, and returning errors if killed, like this?

```
      if (thd && thd->killed) {
        return HA_ERR_QUERY_INTERRUPTED;
      }
```